### PR TITLE
architecture infrastructure: adjust outdated info

### PR DESCRIPTION
**Pull request by chriz-uniba**

### CODE REVIEW

File: docs/develop/architecture/infrastructure.md
Comment (fenekku) [inline docs/develop/architecture/infrastructure.md:14]: TBH support for ElasticSearch v7 is ambiguous. We say it's deprecated, but it will only be removed in a later version (https://inveniordm.docs.cern.ch/releases/versions/version-v11.0.0/#requirements)... so it's still supported: https://github.com/inveniosoftware/cookiecutter-invenio-rdm/blob/master/cookiecutter.json#L17 is my understanding. For now this might be the most accurate:

```suggestion
- **Search engine:** OpenSearch v2, OpenSearch v1, ElasticSearch v7.
```

Then when v12 comes out (I think that's when we drop support or ElasticSearch really) we can have:

```
- **Search engine:** OpenSearch v2, OpenSearch v1.
```

**Review by fenekku — APPROVED** [review]
Thanks. Easy suggestion to get in and after that we can merge.

**Comment by chokribr** [comment]
@kpsherva need one more reviewer to aprove

**Comment by ntarocco** [comment]
Thanks a lot! I am closing this one, we already merged a similar documentation fix.

**Pull request closed without merging**
